src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1971,77 +2000,84 @@ bool isRefEscape
             Debug.Assert(AllParametersConsideredInEscapeAnalysisHaveArguments(argsOpt, parameters, argsToParamsOpt));
 #endif
 
+            MethodInfo localMethodInfo = ReplaceWithExtensionImplementationIfNeeded(methodInfo, ref parameters, ref receiver, ref argsOpt, ref argRefKindsOpt, ref argsToParamsOpt);
+
             if (methodInfo.UseUpdatedEscapeRules)
```

Comment:
>if (methodInfo.UseUpdatedEscapeRules) [](http://example.com/codeflow?start=12&length=37)

Should this check be moved into the local function?

Reply:
I prefer to leave this structure/pattern (`if (newRules) handleNewRules() else handleOldRules()`)

Reply:
> I prefer to leave this structure/pattern (`if (newRules) handleNewRules() else handleOldRules()`)

I wasn't suggesting to change that, and I do not think moving the ```if``` into the local function would change that.